src/resultset/resultset.c

```diff
@@ -26,6 +26,55 @@ static int _encounteredRecord(ResultSet *set, const Record r) {
     return !newRecord;
 }
 
+/* Redis prints doubles with up to 17 digits of precision, which captures
+ * the inaccuracy of many floating-point numbers (such as 7.3).
+ * By using the %g format and its default precision of 6, we avoid awkward
+ * representations like RETURN 7.3 emitting "7.2999999999999998". */
+static inline void _ResultSet_ReplyWithRoundedDouble(RedisModuleCtx *ctx, double d) {
+    // Get length required to print number
+    int len = snprintf(NULL, 0, "%g", d);
+    char str[len + 1]; // TODO a reusable buffer would be far preferable
+    sprintf(str, "%g", d);
+    // Output string-formatted number
+    RedisModule_ReplyWithStringBuffer(ctx, str, len);
+}
```

Comment:
I don't love this function, but see no way to avoid it - doubles are an imperfect representation, as we know too well.

A re-usable buffer would definitely make me a bit happier, but brief testing indicates that our runtimes are generally better with the changes in this PR already.